plugin/pkg/auth/authorizer/node/node_authorizer.go

```diff
@@ -302,8 +302,11 @@ func (r *NodeAuthorizer) hasPathFrom(nodeName string, startingType vertexType, s
 	}
 
 	// Fast check to see if we know of a destination edge
-	if r.graph.destinationEdgeIndex[startingVertex.ID()].has(nodeVertex.ID()) {
-		return true, nil
+	if index := r.graph.destinationEdgeIndex[startingVertex.ID()]; index != nil {
+		if index.has(nodeVertex.ID()) {
+			return true, nil
+		}
+		return false, fmt.Errorf("node %q cannot get %s %s/%s, no relationship to this object was found in the node authorizer graph", nodeName, vertexTypes[startingType], startingNamespace, startingName)
```

Comment:
suggest modifying this message so we know we took the index path

```suggestion
		return false, fmt.Errorf("node %q cannot get %s %s/%s, no relationship to this object was found in the node authorizer graph index", nodeName, vertexTypes[startingType], startingNamespace, startingName)
```